packages/messaging/src/controllers/window-controller.ts

```diff
@@ -86,254 +92,144 @@ export class WindowController extends BaseController {
-    });
+  // Unimplemented
+  onTokenRefresh(): Unsubscribe {
+    return () => {};
   }
 
   /**
-   * This will register the default service worker and return the registration
+   * Creates or updates the default service worker registration.
    * @return The service worker registration to be used for the push service.
    */
-  getSWRegistration_(): Promise<ServiceWorkerRegistration> {
-    if (this.registrationToUse) {
-      return this.waitForRegistrationToActivate_(this.registrationToUse);
+  private async getServiceWorkerRegistration(): Promise<
+    ServiceWorkerRegistration
+  > {
+    if (!this.swRegistration) {
+      try {
+        this.swRegistration = await navigator.serviceWorker.register(
+          DEFAULT_SW_PATH,
+          {
+            scope: DEFAULT_SW_SCOPE
+          }
+        );
+      } catch (e) {
+        throw ERROR_FACTORY.create(ErrorCode.FAILED_DEFAULT_REGISTRATION, {
+          browserErrorMessage: e.message
+        });
+      }
     }
 
-    // Make the registration null so we know useServiceWorker will not
-    // use a new service worker as registrationToUse is no longer undefined
-    this.registrationToUse = null;
-
-    return navigator.serviceWorker
-      .register(DEFAULT_SW_PATH, {
-        scope: DEFAULT_SW_SCOPE
-      })
-      .catch((err: Error) => {
-        throw errorFactory.create(ErrorCode.FAILED_DEFAULT_REGISTRATION, {
-          browserErrorMessage: err.message
-        });
-      })
-      .then((registration: ServiceWorkerRegistration) => {
-        return this.waitForRegistrationToActivate_(registration).then(() => {
-          this.registrationToUse = registration;
-
-          // We update after activation due to an issue with Firefox v49 where
-          // a race condition occassionally causes the service worker to not
-          // install
-          // eslint-disable-next-line @typescript-eslint/no-floating-promises
-          registration.update();
-
-          return registration;
-        });
-      });
+    return this.swRegistration;
   }
 
-  /**
-   * This will return the default VAPID key or the uint8array version of the
-   * public VAPID key provided by the developer.
-   */
-  async getPublicVapidKey_(): Promise<Uint8Array> {
-    if (this.publicVapidKeyToUse) {
-      return this.publicVapidKeyToUse;
+  private async messageEventListener(event: MessageEvent): Promise<void> {
+    if (!event.data?.firebaseMessaging) {
+      // Not a message from FCM
+      return;
     }
 
-    return DEFAULT_PUBLIC_VAPID_KEY;
+    const { type, payload } = (event.data as InternalMessage).firebaseMessaging;
+
+    if (this.onMessageCallback && type === MessageType.PUSH_RECEIVED) {
+      this.onMessageCallback(payload);
+    }
+
+    const { data } = payload;
+    if (
+      data &&
+      FN_CAMPAIGN_ID in data &&
+      data[FN_CAMPAIGN_ANALYTICS_ENABLED] === '1'
+    ) {
+      // This message has a campaign id, meaning it was sent using the FN Console.
```

Comment:
The branding seems to be "Firebase Cloud Messaging" / "the Firebase Console" / "the Notifications composer"

https://firebase.google.com/docs/cloud-messaging/js/send-with-console

So if you want the comment to make sense to someone who doesn't know what "FN" means, I'd just say "the Firebase Console" or "the Notifications composer".

I guess the FN_... constants should just be COMPOSER or CONSOLE?

Also "id" -> "ID" (we aren't talking about https://en.wikipedia.org/wiki/Id,_ego_and_super-ego#Id)

Reply:
"Firebase Console" sounds good. Changed the constants as well.